compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -390,6 +390,12 @@ public BType checkExpr(BLangExpression expr, SymbolEnv env, BType expType, Diagn
         return resultType;
     }
 
+    private void analyzeObjectConstructor(BLangNode node, SymbolEnv env) {
+        if (!nonErrorLoggingCheck) {
+            semanticAnalyzer.analyzeNode(node, env);
```

Comment:
We need to handle other places where we call the semantic analyzer from the type checker too. Please check and create an issue.